Makefile

```diff
@@ -0,0 +1,496 @@
+	  --target aarch64-apple-darwin \
+	  --package core-crypto-ffi \
+	  --crate-type=cdylib --crate-type=staticlib \
+	  --release -- -C strip=symbols
+	@touch $@
+
+.PHONY: jvm-darwin
+jvm-darwin: $(STAMPS)/jvm-darwin
+
+# linux build
+$(STAMPS)/jvm-linux: $(STAMPS)/bindings-kotlin-jvm $(STAMPS)
+	cd crypto-ffi && \
+	$(CARGO) rustc --locked \
+	  --target x86_64-unknown-linux-gnu \
+	  --package core-crypto-ffi \
+	  --crate-type=cdylib --crate-type=staticlib \
+	  --release -- -C strip=symbols
+	@touch $@
+
+.PHONY: jvm-linux
+jvm-linux: $(STAMPS)/jvm-linux
+
+.PHONY: jvm
+ifeq ($(UNAME_S),Linux)
+jvm: jvm-linux
+else ifeq ($(UNAME_S),Darwin)
+jvm: jvm-darwin
+else
+$(error Unsupported host platform for jvm: $(UNAME_S))
+endif
+
+.PHONY: jvm-test
+jvm-test: jvm
+	cd crypto-ffi/bindings && \
+	./gradlew jvm:build -x lint -x lintRelease
+
+
+#-------------------------------------------------------------------------------
+# 9) Aggregate targets
+#-------------------------------------------------------------------------------
+
+.PHONY: mobile wasm all
+mobile: android ios
+wasm: bindings-js
+all: mobile wasm docs
+
+#-------------------------------------------------------------------------------
+# TypeScript / JS tasks (from crypto-ffi/bindings/js/package.json)
+#-------------------------------------------------------------------------------
+
+JS_DIR := crypto-ffi/bindings/js
+JS_SRC_DIR   := $(JS_DIR)/src
+PACKAGE_JSON := $(JS_DIR)/package.json
+
+GEN_DIR := $(JS_SRC_DIR)/autogenerated
+JS_GEN := $(GEN_DIR)/core-crypto-ffi.js
+WASM_GEN := $(GEN_DIR)/core-crypto-ffi_bg.wasm
+
+# find all .ts source files under src/ except in src/autogenerated
+TS_SRCS   := $(shell find $(JS_SRC_DIR) \( -path $(JS_SRC_DIR)/autogenerated -o -path $(JS_SRC_DIR)/corecrypto.d.ts -prune \) -o \( -type f -name '*.ts' \) -print)
+
+JS_OUT := $(JS_DIR)/src/corecrypto.js
+DTS_OUT := $(JS_DIR)/src/corecrypto.d.ts
+
+
+# always remove old outputs
+.PHONY: ts-clean
+ts-clean:
+	@rm -f $(JS_OUT) $(DTS_OUT) \
+	&& rm -rf $(GEN_DIR)
+
+$(STAMPS)/ts-fmt: $(TS_SRCS) $(STAMPS)
+	cd $(JS_DIR) && bun eslint --max-warnings=0 --fix
+	@touch $@
+
+.PHONY: ts-fmt
+ts-fmt: $(STAMPS)/ts-fmt
+
+$(JS_GEN): wasm-build
+$(WASM_GEN): wasm-build
+
+# build corecrypto.js
+$(JS_OUT): $(STAMPS)/bun-deps $(JS_GEN) $(WASM_GEN) $(TS_SRCS) $(PACKAGE_JSON) $(STAMPS)
+	cd $(JS_DIR) && \
+	bun build \
+	  --target browser \
+	  --format esm \
+	  --outfile src/corecrypto.js \
+	  src/CoreCrypto.ts
+	@touch $@
+
+.PHONY: ts-build
+ts-build: $(JS_OUT)
+
+# generate TypeScript defs only when corecrypto.js changed
+$(DTS_OUT): $(JS_OUT) $(TS_SRCS)
+	cd $(JS_DIR) && \
+	bun x dts-bundle-generator \
+	  --project tsconfig.json \
+	  -o src/corecrypto.d.ts \
+	  --no-check \
+	  --export-referenced-types false \
+	  src/CoreCrypto.ts
+	@touch $@
+
+.PHONY: ts
+ts: $(DTS_OUT)
+
+# run WebDriver tests + bun’s built-in tests
+.PHONY: ts-test
+ts-test: ts
+	export CC_TEST_LOG_LEVEL=1 && \
```

Comment:
I would not set `CC_TEST_LOG_LEVEL=1` in the `Makefile`. That should be up to the user.